doc/specs/#1571 - New Tab Menu Customization/#1571 - New Tab Menu Customization.md

```diff
@@ -108,6 +120,42 @@ The "default" new tab menu could be imagined as the following blob of json:
 }
 ```
 
+Alternatively, we could consider something like the following. This would place
+CMD, PowerShell, and all PowerShell cores in the root at the top, followed by
+nested entries for each subsequent dynamic profile generator.
+
+```jsonc
+{
+    "newTabMenu": [
+        { "type":"profile", "profile": "cmd" },
+        { "type":"profile", "profile": "Windows PowerShell" },
+        { "type": "matchProfile", "on": "source", "match": "Microsoft.Terminal.PowerShellCore" }
+        {
+            "type": "folder",
+            "name": "WSL",
+            "entries": [ { "type": "matchProfile", "on": "source", "match": "Microsoft.Terminal.Wsl" } ]
+        },

```

Comment:
"matchProfile" would work well with something like an "autofolder" type.
Something that is only a folder if `entries.length > 1`. Otherwise it's just a regular entry. That way, if you only have a single WSL entry, you save yourself a lot of patience in that context menu. 🙂
I feel like this would be a worthwhile addition to this spec.

Reply:
heck what if we just did that implicitly - a folder with a single `match` in it is treated as this `autofolder`.

Okay no, but maybe there's definitely room for a property here:

`"expand": {"auto"|"always"}`

on `folder`s.

Reply:
I am in favour of the `expand` property, and it is simple to implement in the current setup!